vhost_user_fs: increase RLIMIT_NOFILE

ricarkol opened the pull request:
This change increases the limit of open files for the vhost_user_fs sandboxed process to the maximum allowed in the system. This change is based on this QEMU/virtio-fs commit [1]. One easy way to hit the limit is by creating more than 1024 files (the default is typically 1024) [2] in the vhost_user_fs mount. The last attempts will fail with a "Too many open files" error.

The system maximum is obtained by reading the `/proc/sys/fs/nr_open` sysctl file, and the setting is done using the `setrlimit` syscall. Failure to read or parse the nr_open file, or to set the limit results in a panic.

[1] https://gitlab.com/virtio-fs/qemu/-/commit/ddd21b95215342f5f2d9b94b0960c7292966c075
[2] ```for i in `seq 1 1025`; do touch $i; done```

sboeuf commented on vhost_user_fs/src/sandbox.rs:
```diff
@@ -249,6 +256,22 @@ impl Sandbox {
+            .parse()
+            .map_err(|e| Error::InvalidNrOpen(e))?;
+
+        setrlimit(Resource::NOFILE, limit, limit).map_err(|e| Error::SetRlimit(e))?;
```
I'm totally fine with the patch overall, but I'd like to avoid the introduction of the new crate `rlimit`. Using `libc` should be enough, and long term, I think we could implement both `getrlimit()` and `setrlimit()` abstractions as part of the `vmm-sys-util` crate in Rust-VMM.

ricarkol replied:
Replaced the `rlimit` crate with a `libc::setrlimit` call.

rbradford commented on vhost_user_fs/Cargo.toml:
```diff
@@ -20,4 +20,5 @@ vhost_rs = { git = "https://github.com/cloud-hypervisor/vhost", branch = "dragon
-vmm-sys-util = "0.6.1"
\ No newline at end of file
+vmm-sys-util = "0.6.1"
+rlimit = "0.3.0"
```
Please try and keep these in alphabetical order.

rbradford replied:
But going further. What does using this crate give us over `libc::setrlimit` ?

ricarkol commented:
Thanks for the review @sboeuf and @rbradford. I replaced the `limit` crate with a `libc::setrlimit` call. The crate wasn't doing anything besides the syscall [1].

I squashed the new commit (and force-pushed). Would you have preferred a commit on top of the previous and only squash if it gets merged?

[1] https://github.com/Nugine/rlimit/blob/master/src/lib.rs#L171-L185

russell-islam commented:
@ricarkol  Looks great. Happy to see someone from my FIU Graduate lab.

ricarkol commented:
Hey @russell-islam ! thank you very much!
